src/main/java/de/zalando/zmon/dataservice/proxies/entities/DefaultEntitiesService.java

```diff
@@ -15,57 +17,60 @@
 
 /**
- * 
  * @author jbellmann
- *
  */
 public class DefaultEntitiesService implements EntitiesService {
 
-	
-	private final DataServiceConfigProperties config;
-	private final ObjectMapper customObjectMapper;
-	private final DataServiceMetrics metrics;
+    private final DataServiceConfigProperties config;
+    private final ObjectMapper customObjectMapper;
+    private final DataServiceMetrics metrics;
 
-	public DefaultEntitiesService(@CustomObjectMapper ObjectMapper customObjectMapper, DataServiceConfigProperties config, DataServiceMetrics metrics) {
-		this.customObjectMapper = customObjectMapper;
-		this.config = config;
-		this.metrics = metrics;
-	}
+    public DefaultEntitiesService(@CustomObjectMapper ObjectMapper customObjectMapper, DataServiceConfigProperties config, DataServiceMetrics metrics) {
+        this.customObjectMapper = customObjectMapper;
+        this.config = config;
+        this.metrics = metrics;
+    }
 
-	@Override
-	public String deleteEntity(String id) throws URISyntaxException, IOException{
-        URI uri = new URIBuilder().setPath(config.getProxyControllerUrl() + "/entities/"+id+'/').build();
+    public static HttpClient getHttpClient(int socketTimeout, int timeout, int maxConnections) {
+        RequestConfig config = RequestConfig.custom().setSocketTimeout(socketTimeout).setConnectTimeout(timeout).build();
+        return HttpClients.custom().setMaxConnPerRoute(maxConnections).setMaxConnTotal(maxConnections).setDefaultRequestConfig(config).build();
+    }
 
-        final Executor executor = Executor.newInstance();
+    @Override
+    public String deleteEntity(String id) throws URISyntaxException, IOException {
+        URI uri = new URIBuilder().setPath(config.getProxyControllerUrl() + "/entities/" + id + "/").build();
 
+        final Executor executor = Executor.newInstance(getHttpClient(100, 5000, 1));
         return executor.execute(Request.Delete(uri)).returnContent().asString();
-	}
+    }
 
-	@Override
-	public String getEntities(String query) throws URISyntaxException, IOException {
-		URI uri = new URIBuilder().setPath(config.getProxyControllerUrl() + "/entities/").setParameter("query",query).build();
-        final String r = Request.Get(uri).execute().returnContent().asString();
-        return r;
-	}
+    @Override
+    public String getEntities(String query) throws URISyntaxException, IOException {
+
+        URI uri = new URIBuilder().setPath(config.getProxyControllerUrl() + "/entities/").setParameter("query", query).build();
 
-	@Override
-	public String addEntities(String entities) throws URISyntaxException, IOException {
-		JsonNode node = customObjectMapper.readTree(entities);
-		if (node.has("infrastructure_account")) {
-			String id = node.get("infrastructure_account").textValue();
-			metrics.markEntity(id, 1);
-		}
+        final Executor executor = Executor.newInstance(getHttpClient(100, 5000, 1));
+        return executor.execute(Request.Get(uri)).returnContent().asString();
```

Comment:
there is also a lot of repeated code, executor and maybe execute could be done only once in a method (you already have `getHttpClient` as a helper method anyway)